docs/integration/KafkaIntegration.md

```diff
@@ -20,12 +20,30 @@ Kafka topics are native streaming data input to Nussknacker and the native outpu
 
 ## Schema Registry integration
 
-Nussknacker integrates with Schema Registry. It is the source of topics available to choose from in Kafka sources and sinks. It also allows Nussknacker to provide syntax suggestions and validation. Nussknacker assumes that for the topic `topic-name` a schema `topic-name-value` and optionally `topic-name-key` (for the Kafka topic key) will be defined in the schema registry.
-
-Schemas are stored and managed by Confluent Schema Registry; it is [bundled with Nussknacker](/about/TypicalImplementation) in all deployment versions. Schemas can be registered in Schema Registry by means of REST API based CLI or using AKHQ, an open source GUI for Apache Kafka and Confluent Schema Registry. AKHQ is bundled with Nussknacker in all deployment versions.
-
-Nussknacker supports both JSON and Avro schemas, and JSON and Avro topic payloads. Detailed information on how Avro data should be serialized/deserialized can be found in [Confluent Wire Documentation](https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#wire-format).
-
+Nussknacker integrates with Schema Registries. It is the source of knowledge about:
+- Which topics are available in Kafka sources and sinks
+- What schema versions are available for topic
+- How code completions and validations on types described by schemas should work
+- How messages will be deserialized and serialized
+
+Currently, Nussknacker supports two implementations of schema registries: based on *Confluent Schema Registry* and based on *Azure Schema Registry*. 
+During runtime, it deserializes and serializes messages in the way that is compatible with standard Kafka serializers and deserializers 
+delivered by those schema registries providers. Thanks to that you should be able to send messages to Nussknacker and read messages 
+produced by Nussknacker using standard tooling available around those Schema Registries.
+
+Given implementation is picked based on `schema.registry.url` property. By default, it is used Confluent-based implementation. 
+For urls ended with `.servicebus.windows.net` it is used Azure-based implementation.
+
+### Schema with topic association
+
+To properly present information about topics and version and to recognize which schema is assigned to version, Nussknacker follow conventions:
+- For Confluent-based implementation it uses [TopicNameStrategy for subject names](https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#subject-name-strategy).
+  It means that it looks for schemas available at topic-name-(key or value). For example for topic transactions, it looks for schemas at transactions-key for key and transactions-value subject for value
```

Comment:
`<topic-name>-(key or value)` maybe we can mark somewhow, that topic-name it's a placeholder?

Reply:
placeholders in <> looks better :+1: . Regarding the placeholders description, I don't know how to do this in a way that it won't look too much complex.

Reply:
For me it's understandable, since you provided an example. Altough I would add topic/subject names between ` ` : 

For example for topic `transactions`, it looks for schemas at `transactions-key` for key and `transactions-value` subject for value